### ISSUE TITLE
Incorrect Weight Sharing in MyLinear Class in the Wide_and_Deep model implementation

### ISSUE THREAD

**Issue by shimin3x** [ShowMeAI-Hub/ctr-prediction#2]
## Problem Description

In the current implementation of the `MyLinear` class, a single shared weight matrix is used for all sparse features. This is incorrect because each sparse feature should have its own set of linear weights, as they represent distinct categorical variables that require individual processing.

## Current Implementation Issues

1. **MyLinear Class**:

A single weight matrix self.w of shape (feature_length, 1) is created for all features
This forces all sparse features to share the same weights
```
class MyLinear(Layer):
    # ....
    def build(self, input_shape):
        self.w = self.add_weight(name="w",
                                 shape=(self.feature_length, 1),
                                 regularizer=l2(self.w_r),
                                 trainable=True)
```
2. **Data Processing**:

Each sparse feature is individually label-encoded, and they are distinct categorical variables

Here is the data processing implementation, we can find out that sparse features are transformed separately.
```
    for feat in sparse_features:
        le = LabelEncoder()
        data_df[feat] = le.fit_transform(data_df[feat])
```

## Proposed Fix

The MyLinear class should be modified to create separate weight parameters for each feature rather than using a shared weight matrix.

```
class CustLinear(Layer):
    def __init__(self, sparse_feature_columns, w_r=1e-6):
        super(CustLinear, self).__init__()
        self.sparse_feature_columns = sparse_feature_columns
        self.w_r = w_r
        self.b = None
        self.linear_weights = []

    def build(self, input_shape):
        for i in range(input_shape[-1]):
            feat = self.sparse_feature_columns[i]
            w = self.add_weight(name="w_"+feat['feat_name'],
                                 shape=(feat['feat_num'], 1),
                                 regularizer=l2(self.w_r),
                                 trainable=True)
            self.linear_weights.append(w)
        self.b = self.add_weight(name='bias', shape=(1,), initializer='zeros', trainable=True)

    def call(self, inputs, **kwargs):
        n_feat = inputs.shape[-1]
        weights = []
        for i in range(n_feat):
            weights.append(tf.nn.embedding_lookup(self.linear_weights[i], inputs[:,i]))  # (batch_size, 1)
        return tf.add_n(weights)+self.b
```
